apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -23,16 +33,47 @@ const AnimationPickerBody = React.createClass({
 
   getInitialState() {
     return {
-      searchQuery: ''
+      searchQuery: '',
+      categoryQuery: ''
     };
   },
 
   onSearchQueryChange(value) {
     this.setState({searchQuery: value});
   },
 
+  onCategoryChange(category) {
+    this.setState({categoryQuery: category});
+  },
+
+  onClearCategories() {
+    this.setState({categoryQuery: '', searchQuery: ''});
+  },
+
+  animationCategoriesRendering() {
+    return Object.keys(AnimationCategories).map(category =>
+      <AnimationPickerListItem
+        key={category}
+        label={AnimationCategories[category]}
+        category={category}
+        onClick={() => this.onCategoryChange(category)}
```

Comment:
Might be micro-optimizing, but every time you render this, you'll create a new function object. Another option might be to have `onCategoryChange` extract the category from `event.target`, allowing us to just do `onClick={this.onCategoryChange}`